app/controllers/import_controller.rb

```diff
@@ -322,15 +284,17 @@ def validate_workflow_specific_enums(workflow, field, value, row_ind)
     raise ValidationError.new(err_msg, row_ind)
   end
 
-  def validate_required_primary_contact(patient, row_ind)
-    if patient[:email].blank? && patient[:preferred_contact_method] == 'E-mailed Web Link'
-      raise ValidationError.new("'Email' is required when Primary Contact Method is 'E-mailed Web Link'", row_ind)
-    end
-    unless patient[:primary_telephone].blank? && (['SMS Texted Weblink', 'Telephone call', 'SMS Text-message'].include? patient[:preferred_contact_method])
-      return
-    end
+  def format_model_validation_errors(resource)
```

Comment:
Can we actually put this method on the models or in a helper? That way it doesn't have to duplicated between here, the API controller, and other controllers once we want to validate these for the UI as well.

Reply:
Good call, done here https://github.com/SaraAlert/SaraAlert/pull/582/commits/47f64c407b23f719c346ad7a22e1a2360fb525bd